blog/_posts/2017-08-01-the-search-for-the-killer-app-of-unikernels.md

```diff
@@ -0,0 +1,67 @@
+
+![Steam engine]({{BASE_PATH}}/images/steam_engine.jpg)
+
+The situration with unikernels is similar. We have this new thing and to some extent we're using it to replace some genereal purpose operating system workload. But we're still very much limited by how we think about operating systems and computers.
+
+Unikernels are radically different. Naturally the question of the killer app has come up on a number of occasions. As unikernels are quite different from the dominant operating systems of today it isn’t as easy to spot what the killer application for Unikernels will be. Here I’ll try to answer why this is hard.
+
+## Defining characteristics of Unikernels
+
+Let’s start with what our charactistics are. Unikernels are fast, secure and small. This makes them a good fit for a number of applications. Unikernel Systems original business plan was to pursue serverless architectures. Booting [MirageOS] in a few milliseconds was the driving characteristic behind the initiative. Galios, through their [HaLVM] unikernel are mainly relying on Unikernels being small and frugal to pack a lot of virtual machines into a small little box in their [Cyberchaff] product. [IncludeOS] is focused on network appliances like firewall, load balancers and web based APIs, relying on security and performance to get adoption.
+
+But there is an overlooked characteristic of Unikernels. They are flexible. Unlike Linux and Windows you can easily boot a Unikernel that relies on completely different infrastructure and protocols. Porting something as obscure as SNA to a Unikernel is a lot simpler than porting it to Linux or Windows. Likewise, we could well envision having HTTP support in a unikernel without having an IP stack. This sounds crazy and you’re likely asking yourself why would you ever do such a thing. I’ll get back to why and how, but bear with me for now.
+
+Let us have a look at something the software industry has been struggling with for a while; running native code in browsers. We’ve always known it is a horrible idea but we’ve still spent hundred of thousands of hours making it a reality. If we only were able to build a properly secure sandbox it’ll be great. Turns out making such a sandbox is really hard and this is why [Java Applets], [ActiveX controllers] and [NaCL] have been abandoned (to be truthful, I’m not certain about why Google decided to discontinue NaCL, they’re a bit opaque on why).
+
+## How can Unikernels allow one to run native code in the browser?
+
+It’s reasonably simple. Modern CPUs provide virtualization capabilities built in. These have good security records and to my knowledge an attack that shows how you can break out of the hardware isolation hasn’t be demonstrated. The closest thing we’ve seen is attacks on the hardware emulators that helt facilitate these virtual machines. 
```

Comment:
"These have good security records" -> Unclear what this is referring to, hardware virtualization as a technology?

I can understand what you're getting at -- hardware virt in isolation has a good track record, all attacks to date have been on the "monitor" side -- but this needs more expansion for people outside of the unikernel development community to understand.